docs/eventlisteners.md

```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
 - `namespace` - Refers to the namespace of the EventListener
 - `eventID` - Refers to the uniqueID that gets assigned to each incoming request
 
+## How does the EventListener work?
+
+By now its clear that what are the configuration file for an `EventListener` with `Required` and `Optional` fields explanation.
+
+Lets understand how EventListener works with sample Github example
+
+Create a sample GitHub example
+
+```bash
+kubectl create -f https://github.com/tektoncd/triggers/tree/master/examples/github
+```
+Once creation is success we see a new `Deployment` and `Service` running with the help of `kubectl` command line tool
+```bash
+kubectl get deployment
+NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
+el-github-listener-interceptor   1/1     1            1           11s
+
+kubectl get svc
+NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+el-github-listener-interceptor   ClusterIP   10.99.188.140   <none>        8080/TCP   52s
+```
+It indicates that EventListener code take all information from eventlistener yaml file which are listed in [Syntax](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#syntax), [Lables](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#labels)
+and also hardcode some of the information like `Image, Name, Port` etc... to create **Deployment** and **Service**, So its the responsibility of eventlistener to create Deployment and Service.
```

Comment:
Probably worth adding:

We follow a naming convention when creating these resources. An EventListener named `foo` will create a deployment and a service both named `el-foo`.